### PR TITLE
[WFCORE-5489] Documentation for overridding the value of an attribute with an env var

### PR DESCRIPTION
**Pull request by jmesnil**
JIRA: https://issues.redhat.com/browse/WFCORE-5489

Signed-off-by: Jeff Mesnil <jmesnil@redhat.com>

### CODE REVIEW

**Comment by jmesnil** [comment]
WildFly Core feature is at wildfly/wildfly-core/pull/4639
Dev analysis is at wildfly/wildfly-proposals#404

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 1]
```diff
@@ -285,6 +285,55 @@ See <<descriptions,Descriptions>>
 below for how to learn more about the attributes a particular resource
 exposes.
 
+[[overridden-attribute-value]]
+=== Override Attribute Value with an Environment Variable
+
+It is possible to override the value of any attribute by providing an Environment Variable with a name
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:291]: s/any attribute/any simple attribute/g

And then somewhere a discussion of what a simple attribute is (not a ModelType LIST, OBJECT, PROPERTY.)

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 2]
```diff
@@ -285,6 +285,55 @@ See <<descriptions,Descriptions>>
 below for how to learn more about the attributes a particular resource
 exposes.
 
+[[overridden-attribute-value]]
+=== Override Attribute Value with an Environment Variable
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:289]: s/Override Attribute/Override an Attribute/g

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 3]
```diff
@@ -285,6 +285,55 @@ See <<descriptions,Descriptions>>
...
+
+It is possible to override the value of any attribute by providing an Environment Variable with a name
+that maps to the attribute (and its resource).
+If there is an environment variable with such as name, the management resource will use the value of this environment variable
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:293]: s/such as name/such a name/g

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 4]
```diff
@@ -285,6 +285,55 @@ See <<descriptions,Descriptions>>
...
+when the management resource validate and set the attribute value.
```
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:294]: s/validate and set/validates and sets/g

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc (hunk 1 above)
Comment (jmesnil) [reply docs/src/main/asciidoc/_admin-guide/Management_resources.adoc]: done, thanks

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc (hunk 3 above)
Comment (jmesnil) [reply docs/src/main/asciidoc/_admin-guide/Management_resources.adoc]: fixed

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc (hunk 4 above)
Comment (jmesnil) [reply docs/src/main/asciidoc/_admin-guide/Management_resources.adoc]: fixed

**Review by bstansberry — APPROVED** [review]

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 5]
```diff
@@ -285,6 +285,62 @@ See <<descriptions,Descriptions>>
 below for how to learn more about the attributes a particular resource
 exposes.
 
+[[overridden-attribute-value]]
+=== Override an Attribute Value with an Environment Variable
+
+It is possible to override the value of any simple attribute by providing an environment variable with a name
+that maps to the attribute (and its resource).
+
+[NOTE]
```
Comment (vsalbaba) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:294]: This not does not render correctly in 
https://github.com/wildfly/wildfly/blob/c5b89a2567849cbf3d8334d9655add4f478748b8/docs/src/main/asciidoc/_admin-guide/Management_resources.adoc

It leaves the equals signs in the note. I think we need to use 4 equal signs here

Comment (jmesnil) [reply docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:294]: fixed, thanks!

**Comment by vsalbaba** [comment]
LGTM

**Comment by jmesnil** [comment]
Dev PR has been merged at https://github.com/wildfly/wildfly-core/pull/4816. This doc PR can be merged when the next wildfly-core release is incorporated in WildFly

File: docs/src/main/asciidoc/_admin-guide/Management_resources.adoc [hunk 6]
```diff
@@ -285,6 +285,62 @@ See <<descriptions,Descriptions>>
...
+====
+Complex attributes (which have their `type` set to `LIST, OBJECT`, or `PROPERTY`) can not be overridden using an environment variable.
```
Comment (rhusar) [inline docs/src/main/asciidoc/_admin-guide/Management_resources.adoc:296]: `LIST, OBJECT` -> `LIST`, `OBJECT`

**Pull request merged**